### PR TITLE
TensorFloat32 Test Fixes for TensorFlow core and python tests

### PR DESCRIPTION
**Pull request by nluehr**
Configure tests to use float32 compute type as expected by the thresholds as currently calibrated.

Attn: @reedwm

### CODE REVIEW

**Review by reedwm — APPROVED** [review]

File: tensorflow/compiler/xla/tests/convolution_variants_test.cc [hunk 1]
```diff
@@ -34,6 +34,7 @@ limitations under the License.
 #include "tensorflow/compiler/xla/tests/literal_test_util.h"
 #include "tensorflow/compiler/xla/tests/test_macros.h"
 #include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "tensorflow/tsl/platform/tensor_float_32_utils.h"
```
Comment (akuegel) [inline tensorflow/compiler/xla/tests/convolution_variants_test.cc:37]: This header include needs a corresponding BUILD file change.

**Review by akuegel — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/tests/convolution_variants_test.cc (hunk 1 above)
Comment (nluehr) [reply tensorflow/compiler/xla/tests/convolution_variants_test.cc:37]: I've updated the BUILD file with the following, which seems to work to build both CPU and GPU tests.
```
select({
        "//tensorflow/tsl:is_cuda_enabled": [
            "//tensorflow/tsl/platform:tensor_float_32_hdr_lib",
        ],
        "//conditions:default": [
            "//tensorflow/tsl/platform:tensor_float_32_utils",
        ],
    }),
```

**Review by akuegel — APPROVED** [review]

**Comment by nluehr** [comment]
@akuegel is there anything I can do to get the merge of this PR unblocked?

**Comment by reedwm** [comment]
I am attempting to fix internal failures and get this merged

**Comment by reedwm** [comment]
@nluehr in the merge commit I changed the `convolution_variants_test` rule to unconditionally depend on `//tensorflow/tsl/platform:tensor_float_32_utils` instead of just when CUDA is enabled. I did this because internally the `xla_test` internally iterated over `deps` and therefore did not like the `select` call you added. Let me know if there are any issues with unconditionally depending on `//tensorflow/tsl/platform:tensor_float_32_utils`. I tested the test passes on Ampere both externally and internally.

**Pull request merged**